source/extensions/outlier_detection_monitors/consecutive_errors/consecutive_errors_monitor.cc

```diff
@@ -0,0 +1,56 @@
+#include "source/extensions/outlier_detection_monitors/consecutive_errors/consecutive_errors_monitor.h"
+
+#include "envoy/extensions/outlier_detection_monitors/consecutive_errors/v3/consecutive_errors.pb.h"
+#include "envoy/extensions/outlier_detection_monitors/consecutive_errors/v3/consecutive_errors.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Outlier {
+
+bool ConsecutiveErrorsMonitor::onError() {
+  uint32_t expected_count = counter_.load();
+
+  // no-op. Just keep executing compare_exchange_strong until threads synchronize.
+  do {
+    ;
+  } while (!counter_.compare_exchange_strong(expected_count, expected_count + 1));
```

Comment:
i think the [default memory order](https://en.cppreference.com/w/cpp/atomic/atomic/compare_exchange) `memory_order_seq_cst` here requires too much synchronization across worker threads, you could use relaxed order for cas failure and make writes to atomic visible to other threads on success (e.g. memory order release)

Reply:
How about just `couter_++`?

Reply:
The reason is that while `counter_` can be "atomically" increased, what we need is entire logic of incrementing and checking a value to be deterministic. This approach seems to be sort of pattern for this lock-less approach. See https://github.com/envoyproxy/envoy/pull/34154#discussion_r1611416745